src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs

```diff
@@ -94,12 +96,21 @@ private static IEnumerable<DictionaryEntry> AzureEnvToAppEnv(DictionaryEntry ent
             {
                 prefix = CustomPrefix;
             }
-            else
+            else if (key.StartsWith(envPrefix, StringComparison.OrdinalIgnoreCase))
             {
-                entry.Key = NormalizeKey(key);
+                // This prevents the prefix from being normalized.
+                // We can also do a fast path branch, I guess? No point in reallocating if the prefix is empty.
+                entry.Key = envPrefix != string.Empty
```

Comment:
if the key starts with the `envPrefix` when would `envPrefix` be `string.Empty`?

Reply:
In the event user does not specify a prefix, it defaults to `string.Empty`. Obviously, for any given string s such that s is not null, `s.StartsWith(string.Empty, StringComparison.OrdinalIgnoreCase)` is true.

Reply:
See lines 38 and 29 for where that assumption comes from.

Reply:
Makes sense. It kind of confused me but now that I see this it makes sense.